pkg/webhook/board_update.go

```diff
@@ -60,16 +78,91 @@ func (h *boardUpdate) handleIssuesEvent(event *github.IssuesEvent, gh *github.Cl
 	logger.Debug("Issue Action: " + *event.Action)
 
 	eventKey := messageType + "_" + *event.Action
+
+	// post processing (from previous event cycle)
+	// takes precedence, the event will no be processed further
+	if "issues_closed" == eventKey {
+		_, issue_scheduled := postProcessing[number]
+
+		if issue_scheduled {
+			delete(postProcessing, number)
+			go postProcess(event, gh, config, *doneColumn, logger)
+			return nil
+		}
+	} else if "issues_reopened" == eventKey && event.GetIssue().GetLocked() {
+		// move
+		err := moveIssueOnBoard(config, number, *doneColumn, logger)
+
+		// update progress/* label
+		changeProgressLabel(gh, event.Repo, *event.Issue, doneColumn.name)
+
+		if err != nil {
+			logger.Error("Post processing failed: Cannot move issue")
+		}
+	}
+
+	// cleanup post processing markers, but skip actions
+	if event.GetIssue().GetLocked() {
+		gh.Issues.Unlock(context.Background(), event.Repo.Owner.GetLogin(), event.Repo.GetName(), *event.Issue.Number)
+		logger.Debug("Unlock and ignore event for: " + number)
+		return nil
+	}
+
+	// regular processing
 	col, ok := stateMapping[eventKey]
 	if ok {
-		return moveIssueOnBoard(config, number, col, logger)
+		err := moveIssueOnBoard(config, number, col, logger)
+
+		if nil == err {
+			// update progress/* label
+
+			changeProgressLabel(gh, event.Repo, *event.Issue, col.name)
+		}
+
+		return err
 	} else {
-		logger.Debug("Ignore unmapped event: " + eventKey)
+		logger.Debug("Ignore unmapped Issue event: " + eventKey)
 	}
 
 	return nil
 }
 
+func changeProgressLabel(gh *github.Client, repo *github.Repository, issue github.Issue, newLabel string) {
+
+	labels := []string{"progress/" + newLabel}
+
+	for _, label := range issue.Labels {
+		if strings.HasPrefix(*label.Name, "progress/") {
+			gh.Issues.RemoveLabelForIssue(context.Background(), repo.Owner.GetLogin(), repo.GetName(),
+				issue.GetNumber(), *label.Name)
+
+		}
+	}
+
+	gh.Issues.AddLabelsToIssue(context.Background(), repo.Owner.GetLogin(), repo.GetName(),
+		issue.GetNumber(), labels)
+}
+
+func postProcess(event *github.IssuesEvent, gh *github.Client, config config.RepoConfig, col column, logger *zap.Logger) {
+
+	number := strconv.Itoa(*event.Issue.Number)
+
+	logger.Debug("Handle grace time ... ")
+	time.Sleep(10 * time.Second)
+
+	_, e := gh.Issues.Lock(context.Background(), event.Repo.Owner.GetLogin(), event.Repo.GetName(), *event.Issue.Number, &github.LockIssueOptions{LockReason: "resolved"})
+	if e != nil {
+		logger.Error("Locking issue failed: " + number)
+	}
+
+	// re-open
+	state := "open"
+	_, _, err := gh.Issues.Edit(context.Background(), event.Repo.Owner.GetLogin(), event.Repo.GetName(), *event.Issue.Number, &github.IssueRequest{State: &state})
+	if err != nil {
+		logger.Error("Post processing failed ")
+	}
```

Comment:
I think you can remove the `col` method argument here as it is not used (but only when the reopen event is evaluated)

Reply:
+1